securedrop/secure_tempfile.py

```diff
@@ -83,7 +89,12 @@ def write(self, data):
             raise AssertionError('You cannot write after reading!')
         self.last_action = 'write'
 
-        if isinstance(data, str):  # noqa
+        # This is the old Python related code
+        if six.PY2:  # noqa
```

Comment:
This is overly complicated for python2/3 compatibility. The code should be:

```python
if isinstance(data, six.binary_type):
    data = data.decode('utf-8')
```

Reply:
This did not work for me, I just kept the old code for Python2 branch. We are planning to drop python2 support as soon as possible, and we will remove all `six` for the same.